ambry-frontend/src/main/java/com/github/ambry/frontend/AmbrySecurityService.java

```diff
@@ -394,7 +395,8 @@ private void setGetBlobResponseHeaders(BlobInfo blobInfo, GetBlobOptions options
       restResponseChannel.setHeader(RestUtils.Headers.CONTENT_RANGE, rangeAndLength.getFirst());
       contentLength = rangeAndLength.getSecond();
     }
-    if (contentLength < frontendConfig.chunkedGetResponseThresholdInBytes) {
+    if (contentLength < frontendConfig.chunkedGetResponseThresholdInBytes || RestUtils.isS3Request(restRequest)) {
```

Comment:
I would prefer to add content length header in the S3GetBlobHandler so we don't add any special cases here.

Reply:
Sure. Updated to have the logic in S3GetBlobHandler